apps/ybal/components/CardTransactorWrapper.tsx

```diff
@@ -0,0 +1,298 @@
+			toAddress(selectedOptionFrom.value),
+			selectedOptionFrom.zapVia,
+			ethers.constants.MaxUint256
+		).onSuccess(async (): Promise<void> => {
+			await refresh();
+		}).perform();
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** BAL token require the allowance to be reset to 0 before being able to
+	** increase it. This function is called when the user wants to increase the
+	** allowance of the BAL token.
+	**************************************************************************/
+	async function	onIncreaseBalAllowance(): Promise<void> {
+		await new Transaction(provider, approveERC20, set_txStatusApprove).populate(
+			toAddress(selectedOptionFrom.value),
+			selectedOptionFrom.zapVia,
+			0
+		).perform();
+
+		new Transaction(provider, approveERC20, set_txStatusApprove).populate(
+			toAddress(selectedOptionFrom.value),
+			selectedOptionFrom.zapVia,
+			ethers.constants.MaxUint256
+		).onSuccess(async (): Promise<void> => {
+			await refresh();
+		}).perform();
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Execute a zap using the ZAP contract to migrate from a token A to a
+	** supported token B.
+	**************************************************************************/
+	async function	onZap(): Promise<void> {
+		dismissAllToasts();
+
+		const addToMetamaskToast = {
+			type: 'info' as const,
+			content: `Add ${selectedOptionTo.symbol} to Metamask?`,
+			duration: Infinity,
+			cta: {
+				label: 'Add +',
+				onClick: (): void => addToken({
+					address: selectedOptionTo.value,
+					symbol: selectedOptionTo.symbol,
+					decimals: selectedOptionTo.decimals,
+					image: selectedOptionTo.icon?.props.src
+				})
+			}
+		};
+
+		if (selectedOptionFrom.zapVia === LPYBAL_TOKEN_ADDRESS) {
+			// Direct deposit to vault from bal/yB Curve LP Token to lp-yB Vault
+			new Transaction(provider, deposit, set_txStatusZap).populate(
+				toAddress(selectedOptionTo.value), //destination vault
+				amount.raw //amount_in
+			).onSuccess(async (): Promise<void> => {
+				set_amount(toNormalizedBN(0));
+				await refresh();
+				toast(addToMetamaskToast);
+			}).perform();
+		} else {
+			// Zap in
+			new Transaction(provider, zap, set_txStatusZap).populate(
+				toAddress(selectedOptionFrom.value), //_input_token
+				toAddress(selectedOptionTo.value), //_output_token
+				amount.raw, //amount_in
+				expectedOut, //_min_out
+				slippage
+			).onSuccess(async (): Promise<void> => {
+				set_amount(toNormalizedBN(0));
+				await refresh();
+				toast(addToMetamaskToast);
+			}).perform();
+		}
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Set of memorized values to limit the number of re-rendering of the
+	** component.
+	**************************************************************************/
+	const	fromVaultAPY = useMemo((): string => {
+		if (toAddress(selectedOptionFrom.value) === STYBAL_TOKEN_ADDRESS) {
```

Comment:
Wonder if these kinds of comparisons should be case insensitive, e.g.
```suggestion
		if (toAddress(selectedOptionFrom.value).toUpperCase() === STYBAL_TOKEN_ADDRESS.toUpperCase()) {
```

Reply:
Are we sure? Most of the times we're simply casting to `TAddress`, even though they might not be